File: CPP/Program/Komputer.cpp
#pragma once
#include <bits/stdc++.h>
#include "Cpu.cpp"
#include "Ram.cpp"
#include "GPU.cpp"
#include "Harddrive.cpp"
using namespace std;

class Komputer {
    // Attributes
    private:
        string nama;
        Cpu cpu;
        vector<Ram> listRam;
        GPU gpu;
        Harddrive storage;
    
    // Constructor
    public:
        // Default constructor
        Komputer() {
            this->nama = "";
        }
        
        // Parameterized constructor
        Komputer(string nama, Cpu cpu, GPU gpu, Harddrive storage) {
            this->nama = nama;
            this->cpu = cpu;
            this->gpu = gpu;
            this->storage = storage;
        }
        
        // Getters
        string getNama() {
            return this->nama;
        }
        
        Cpu getCpu() {
            return this->cpu;
        }
        
        const vector<Ram>& getListRam () const{
            return this->listRam;
        }
        
        GPU getGpu() {
            return this->gpu;
        }
        
        Harddrive getStorage() {
            return this->storage;
        }
        
        // Setters
        void setNama(string nama) {
            this->nama = nama;
        }
        
        void setCpu(Cpu cpu) {
            this->cpu = cpu;
        }
        
        void setGpu(GPU gpu) {
            this->gpu = gpu;
        }
        
        void setStorage(Harddrive storage) {
            this->storage = storage;
        }
        
        // Methods
        void addRam(Ram ram) {
            this->listRam.push_back(ram);
        }
        
        // Destructor
        ~Komputer() {
        }
};

File: Python/Program/Komputer.py
from Cpu import Cpu
from GPU import GPU
from Harddrive import Harddrive

class Komputer:
    # Constructor
    def __init__(self, nama="", cpu=None, gpu=None, storage=None):
        self.nama = nama
        self.cpu = cpu if cpu is not None else Cpu()
        self.list_ram = []
        self.gpu = gpu if gpu is not None else GPU()
        self.storage = storage if storage is not None else Harddrive()
    
    # Getters
    def get_nama(self):
        return self.nama
    
    def get_cpu(self):
        return self.cpu
    
    def get_list_ram(self):
        return self.list_ram
    
    def get_gpu(self):
        return self.gpu
    
    def get_storage(self):
        return self.storage
    
    # Setters
    def set_nama(self, nama):
        self.nama = nama
    
    def set_cpu(self, cpu):
        self.cpu = cpu
    
    def set_gpu(self, gpu):
        self.gpu = gpu
    
    def set_storage(self, storage):
        self.storage = storage
    
    # Methods
    def add_ram(self, ram):
        self.list_ram.append(ram)
